webapp/pages/settings/my-social-media.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+  <ds-card :header="$t('settings.social-media.name')">
+    <ds-space
+      v-if="socialMediaLinks"
+      margin-top="base"
+      margin="x-small"
+    >
+      <ds-list>
+        <ds-list-item
+          v-for="link in socialMediaLinks"
+          :key="link.url"
+        >
+          <a :href="link.url">
+            <img
+              :src="link.favicon"
+              alt="Social Media link"
+              width="16"
+              height="16"
+            >
+            {{ link.url }}
+          </a>
+        </ds-list-item>
+      </ds-list>
+    </ds-space>
+    <div>
+      <ds-input
+        v-model="value"
+        placeholder="Add social media url"
+        name="social-media"
+        :schema="{type: 'url'}"
+      />
+    </div>
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-03-26T22:03:19Z" title="Tuesday, March 26th 2019, 11:03:19 pm +01:00">Mar 26, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,73 @@
+<template>
+  <ds-card :header="$t('settings.social-media.name')">
+    <div>
+      <ds-input
+        v-model="value"
+        placeholder="Add social media url"
+        name="social-media"
+        :schema="{type: 'url'}"
+      />
+    </div>
+    <ds-space margin-top="base">
+      <div>
+        <ds-button
+          primary
+          @click="handleAddSocialMedia"
+        >
+          {{ $t('settings.social-media.submit') }}
+        </ds-button>
+      </div>
+    </ds-space>
+    <ds-space 
+      v-if="currentUser.socialMedia && currentUser.socialMedia.length"
+      margin-top="base"
+      margin="x-small"
+    >
+      <div
+        v-for="socialMediaIconUrl in currentUser.socialMedia"
+        :key="socialMediaIconUrl"
+      >
+        <a>
+          <img
+            :src="socialMediaIconUrl.match(/^(?:https?:\/\/)?(?:[^@\n])?(?:www\.)?([^:\/\n?]+)/g)[0] + '/favicon.ico'"
```


I like the idea but do you know what happens if this doesn't work?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-03-26T22:16:32Z" title="Tuesday, March 26th 2019, 11:16:32 pm +01:00">Mar 26, 2019</time>_

---


yes, if there is some url that is shared, which does not have an endpoint `/favicon.ico`, then it will display a "broken" image